src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -209,12 +291,229 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+            }
+
+            return variantDefinition;
+        }
+
+        private ValueTask<VariantDefinition> AssignVariantAsync(FeatureDefinition featureDefinition, TargetingContext targetingContext, CancellationToken cancellationToken)
+        {
+            VariantDefinition variant = null;
+
+            if (featureDefinition.Allocation.User != null)
+            {
+                foreach (UserAllocation user in featureDefinition.Allocation.User)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext.UserId, user.Users, _assignerOptions.IgnoreCase))
+                    {
+                        if (string.IsNullOrEmpty(user.Variant))
+                        {
+                            _logger.LogWarning($"Missing variant name for user allocation in feature {featureDefinition.Name}");
+
+                            return new ValueTask<VariantDefinition>((VariantDefinition)null);
+                        }
+
+                        Debug.Assert(featureDefinition.Variants != null);
+
+                        return new ValueTask<VariantDefinition>(
+                            featureDefinition
+                                .Variants
+                                .FirstOrDefault((variant) => variant.Name == user.Variant));
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Group != null)
+            {
+                foreach (GroupAllocation group in featureDefinition.Allocation.Group)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext.Groups, group.Groups, _assignerOptions.IgnoreCase))
+                    {
+                        if (string.IsNullOrEmpty(group.Variant))
+                        {
+                            _logger.LogWarning($"Missing variant name for group allocation in feature {featureDefinition.Name}");
+
+                            return new ValueTask<VariantDefinition>((VariantDefinition)null);
+                        }
+
+                        Debug.Assert(featureDefinition.Variants != null);
+
+                        return new ValueTask<VariantDefinition>(
+                            featureDefinition
+                                .Variants
+                                .FirstOrDefault((variant) => variant.Name == group.Variant));
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Percentile != null)
+            {
+                foreach (PercentileAllocation percentile in featureDefinition.Allocation.Percentile)
+                {
+                    if (TargetingEvaluator.IsTargeted(
+                        targetingContext,
+                        percentile.From,
+                        percentile.To,
+                        _assignerOptions.IgnoreCase,
+                        featureDefinition.Allocation.Seed))
```

Comment:
If seed is null, we should use feature name.

Reply:
This would ensure that flags have different seeds by default. This ensures whoever gets mapped to 1% isn't thrown into every experiment always.

Reply:
Although, there is an argument that Allocation might not want to default to the same % mapping that targeting uses. Because a targeting filter of 50% with an allocation of 50% means 100% of the 50% would be allocated. 

So perhaps we should default to feature name, but add "allocation" at the end or something. @jimmyca15 what do you think?

Reply:
@rossgrambo makes sense. Although I would put any addition before the flag name since it's unchanging and acts like prefix in my eyes.